### ISSUE TITLE
Agree definition of done for view_component migrations and set up code base for work to begin

### ISSUE THREAD

**Issue by kimadactyl** [geeksforsocialchange/PlaceCal#2641]
## User story

As a developer I want a clear definition of done for migrating to view_component so that everything is then completely standardised.

## Acceptance criteria (WIP)

- [ ] A new PR template is created for components that contains the below
- [ ] Components have a `_component` suffix
- [ ] Old mountain_view code is deleted/migrated
- [ ] Evaluate if each component has the best name, or if there could be a terser or more semantic one
- [ ] [Generators](https://viewcomponent.org/guide/generators.html) are configured to create as much of this as possible
- [ ] Stimulus is configured to load in the JS files from the subdirectories
- [ ] Scss is left as is for now and will be reviewed in the future
- [ ] Locale information is added to the global locale file and not stored with the component
- [ ] There are minimal unit tests which do not rely on VCR
- [ ] Light refactoring is encouraged if possible but priority is to get this job done and move on
- [ ] Simple components with a template should appear as below in `/app/components/...`

```
components
├── example_component.rb            # Component definition
└── example_component.html.erb      # HTML template
```

- [ ] More complex components with javascript and scss can be moved into a folder:

```
components
├── example_component.rb            # Component definition
└── example_component
    ├── example_component.html.erb  # HTML template
    ├── example_component.js        # Optional: Stimulus controller
    └── example_component.scss      # Optional: Component-namespaced scss
```

## Implementation notes & questions

- [x] Do we want the `_component` suffix on all of these? Without is neater, with is more explicit
- [x] How do we load in all the scss properly with depends and the like?
- [x] Do we want a locale file per component or just add to the global one?

## Implementation plan

_To be written by the developer_

<sub><a href="https://huly.app/guest/geeksforsocialchange?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJsaW5rSWQiOiI2NzJmNjUxM2U3Y2M1YTc4NTQyOWU0NzgiLCJndWVzdCI6InRydWUiLCJlbWFpbCI6IiNndWVzdEBoYy5lbmdpbmVlcmluZyIsIndvcmtzcGFjZSI6Incta2ltLWdlZWtzZm9yc29jaS02NzFlNzVmOS03ZTVlMTU1YzMwLTIxZjkwZCJ9.5J1CsSuZgRV1oL2_7fpNZWmb1i0x0ET441mDE1ECNzQ">Huly&reg;: <b>PC-2646</b></a></sub>

**Comment by kimadactyl** [comment]
@lenikadali - i thought would be a good idea to hash this out totally before we start moving stuff, what do you think?

**Comment by lenikadali** [comment]
@kimadactyl I think this is a great idea. Will follow up on Discord :innocent:

**Comment by kimadactyl** [comment]
Another thought: given this is a really short template maybe we should consider having it inline in the model file, and set a guideline for when to do inline and when to break out into a folder?

https://viewcomponent.org/guide/templates.html#inline

**Comment by kimadactyl** [comment]
It looks like the official reccomendation is now to **not** have a component suffix but not to sweat it too much - so given we are taking it from the top i suggest we remove it?

https://viewcomponent.org/adrs/0002-naming-conventions-for-view-components.html

**Comment by lenikadali** [comment]
Sample codebase I looked in my research was the [RailsDevs codebase](https://github.com/joemasilotti/railsdevs.com/blob/main/app/components/). I would stick to what `view_component` does out of the box and then if we run into problems that can/need to be solved with a more custom approach, customize incrementally.

**Comment by kimadactyl** [comment]
Updated following discord conversation and now locked in i think!

**Comment by kimadactyl** [comment]
Updated following final decision in Discord to have components in root folder.

**Comment by kimadactyl** [comment]
Do you wanna take on this one next @lenikadali? Hopefully after that these can go _a lot_ faster :D

**Comment by lenikadali** [comment]
> Do you wanna take on this one next @lenikadali? Hopefully after that these can go _a lot_ faster :D

Not sure what the outcome of this ticket should be. My understanding is that between the comments in the [first PR](https://github.com/geeksforsocialchange/PlaceCal/pull/2638) for [this issue](https://github.com/geeksforsocialchange/PlaceCal/issues/2270), we've more or less agreed that definition of done is:

1. Changing the components to use ViewComponent
2. Write unit tests for each migrated component
3. Update calls across the codebase to use the ViewComponent style instead of the `mountain_view` style rendering call
4. Fix any tests that get broken in the process
5. If all pipelines are green, merge into `main`.

Let me know if I've left something out :innocent:

**Comment by kimadactyl** [comment]
Oh just thought there could be a PR template specifically for these checkboxes that was all! Maybe overkill?

**Comment by lenikadali** [comment]
> Oh just thought there could be a PR template specifically for these checkboxes that was all! Maybe overkill? So a "New or updated component" PR template

Ah... I see what you mean :slightly_smiling_face: We can look into how to handle having multiple PR templates and how each gets triggered depending on the issue?

It _may_ be overkill but the [migration issue](https://github.com/geeksforsocialchange/PlaceCal/issues/2270) has a lot of detail that is easy to overlook. Alternatively the issue can be converted into an epic and then when migrating a particular component, the developer makes **_both_** the issue and the PR.

From what I can see, hovering over the checkboxes for the different components shows the option to convert to an issue (have done it as a test [here](https://github.com/geeksforsocialchange/PlaceCal/issues/2646)). So the next step would be to update it with all the detail concerning this particular component's migration and then it should be ready to work on. What do you think?

**Comment by kimadactyl** [comment]
Oh I assumed it'd be like issues, that we can have multiple templates (like feature/bug/tech debt we have now) but I guess it's not quite that simple for PRs, hmm!

Yes I think this is a really nice to have and it'll speed up the process a lot. I think the PR makes more sense to have this checklist than the issues now we've agreed the steps! A quick google indicates this should be possible but maybe with some faff :D

**Comment by lenikadali** [comment]
> tl;dr: issue templates and PR templates are different but i think we should have a PR template if its possible 🙏

Cool. Will take a look and see if I can open an PR for this.

**Comment by kimadactyl** [comment]
ok great. so maybe better we do this as part of https://github.com/geeksforsocialchange/PlaceCal/issues/2551 then?

**Comment by kimadactyl** [comment]
Cool! Another option could be to chuck an .md file in the `components` directory actually?

Like this one: https://github.com/geeksforsocialchange/PlaceCal/blob/main/app/jobs/creating_an_importer.md

**Comment by lenikadali** [comment]
> Cool! Another option could be to chuck an .md file in the `components` directory actually?
> 
> Like this one: https://github.com/geeksforsocialchange/PlaceCal/blob/main/app/jobs/creating_an_importer.md

That should work :raised_hands:

**Comment by lenikadali** [comment]
PR for component guidelines is [here](https://github.com/geeksforsocialchange/PlaceCal/pull/2648)
